logprep/abc/getter.py

```diff
@@ -0,0 +1,24 @@
+"""module for getter interface"""
+from abc import ABC, abstractmethod
+
+from attrs import define, field, validators
+
+
+@define(kw_only=True)
+class Getter(ABC):
+    """abstract base class describing the getter interface and providing of factory method"""
+
+    protocol: str = field(validator=validators.instance_of(str))
+    """indicates the protocol for the factory to chose a matching getter"""
+    target: str = field(validator=validators.instance_of(str))
+    """the target which holds the content to return by get method"""
+
+    @abstractmethod
+    def get(self) -> str:
```

Comment:
no... I disagree Parsing should not be the responsibility of the getter. In addition, it would break backwards compatibility, because you had to describe it also for file based config artifacts in the first part of the URI. (like: `yaml+file://your-file.yaml`. and it is not possible to parse the file name, because this is filesystem dependend and will not work for http (like `yaml+http:/my-api-endpoin/`

Reply:
I wouldn't have put it in the string, I just thought of a python parameter for the developer. So you could do something like:
```
self._regex_mapping = GetterFactory.from_string(regex_mapping_path).get(format="yaml")
```
instead of
```
content = GetterFactory.from_string(regex_mapping_path).get()
self._regex_mapping = yaml.load(content)
```
and if no format is given you just return the raw content, not parsed.

Reply:
This would break the clear interface `string -> Getter -> get() -> string` ... the output of the get method would return other types depending on the parameters. I like python a lot that it could do it, but the possibility it could be done doesn't mean that we have to do it. I prefer to see how the string data is parsed, where it is parsed.

Beyond this, it would lead to decreasing performance, because the resulting if statements have to be evaluated every time, the get method is called. Ok... we could implement it with extra methods, like .get_yaml() or .get_json()... I will check, if this is a good idea.

Reply:
The interface argument I take, but the performance of the `if` statements can be ignored I would say. That's always O(1) after all. :wink: I do like the idea of `get_yaml` or `get_json` though :)

Reply:
I implemented the last idea and refactored everywhere where I had not to touch the tests